cpp/dolfinx/refinement/ParallelRefinement.cpp

```diff
@@ -25,26 +25,43 @@ using namespace dolfinx::refinement;
 
 //-----------------------------------------------------------------------------
 ParallelRefinement::ParallelRefinement(const mesh::Mesh& mesh)
-    : _mesh(mesh), _marked_edges(mesh.num_entities(1), false),
-      _marked_for_update(MPI::size(mesh.mpi_comm()))
+    : _mesh(mesh), _marked_edges(mesh.num_entities(1), false)
```

Comment:
Can we avoid `mesh.num_entities`? I'd like to remove it soon in favour of going via `topology` or `geometry` so that it's clear what's being requested.